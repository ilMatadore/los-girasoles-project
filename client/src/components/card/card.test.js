import React from 'react';
import { shallow } from 'enzyme';
import Card from './card.components';

let wrapper;

beforeEach(() => {
    const mockProps = {
        canasta: {
            name: 'Test Canasta',
            price: 200,
            description: [
                { id: 1, name: 'Perejil', cantidad: '1at'},
                { id: 2, name: 'Mizuna', cantidad: '1'}
        ]
        },
        addItem: jest.fn()         
    }
    wrapper =  shallow(<Card {...mockProps}/>)
})

it('Expect to render Card component', () => {
    expect(wrapper).toMatchSnapshot();   
})

it('Test Button', () => {
    wrapper.find(".card-cart-icon").simulate('click')
    expect(wrapper.length).toEqual(1)
})

