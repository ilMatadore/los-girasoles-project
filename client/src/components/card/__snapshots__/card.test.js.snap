// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Card component 1`] = `
<div
  className="card-container"
  data-aos="flip-right"
>
  <p
    className="card-title"
  >
    TEST CANASTA
  </p>
  <img
    alt="vegetables"
    height="150"
    src="vegetables.png"
    width="150"
  />
  <div
    className="card-price"
  >
    $
    200
  </div>
  <div>
    <ul>
      <li
        key="1"
      >
        PEREJIL
         
        <span>
          - 
          1at
        </span>
      </li>
      <li
        key="2"
      >
        MIZUNA
         
        <span>
          - 
          1
        </span>
      </li>
    </ul>
  </div>
  <button
    className="card-button"
  >
    <FaCartPlus
      className="card-cart-icon"
      onClick={[Function]}
    />
  </button>
</div>
`;
